### ISSUE TITLE
progress dialog not shown while compiling

### ISSUE THREAD

**Issue by pelikhan** [Microsoft/pxt#338]
On slow systems, compilation takes several seconds (@mmoskal). We currently only show the modal dialog once compilation is done. This allows the user to button smash the download button...

(Perfomance issue moved to separate issue)

**Comment by mmoskal** [comment]
Could we have separate issues for speeding up compilation and improving the UI?

**Comment by pelikhan** [comment]
Opened #350
